akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala

```diff
@@ -523,47 +622,45 @@ final class CoordinatedShutdown private[akka] (
         remainingPhases match {
           case Nil => Future.successful(Done)
           case phase :: remaining if !phases(phase).enabled =>
-            tasks.get(phase) match {
-              case null => // This pretty much is ok as there are no tasks
-              case tasks =>
-                log.info("Phase [{}] disabled through configuration, skipping [{}] tasks", phase, tasks.size)
+            tasks.get(phase).foreach { phaseDef =>
+              log.info("Phase [{}] disabled through configuration, skipping [{}] tasks", phase, phaseDef.size)
             }
             loop(remaining)
           case phase :: remaining =>
             val phaseResult = tasks.get(phase) match {
-              case null =>
+              case None =>
                 if (debugEnabled) log.debug("Performing phase [{}] with [0] tasks", phase)
                 Future.successful(Done)
-              case tasks =>
-                if (debugEnabled)
+              case Some(phaseDef) =>
+                if (debugEnabled) {
                   log.debug(
                     "Performing phase [{}] with [{}] tasks: [{}]",
                     phase,
-                    tasks.size,
-                    tasks.map { case (taskName, _) => taskName }.mkString(", "))
-                // note that tasks within same phase are performed in parallel
+                    phaseDef.size,
+                    phaseDef.values.flatten.mkString(", "))
+                }
                 val recoverEnabled = phases(phase).recover
                 val result = Future
-                  .sequence(tasks.map {
-                    case (taskName, task) =>
+                  .sequence(phaseDef.map {
+                    case (task, names) =>
                       try {
-                        val r = task.apply()
+                        val r = task()
                         if (recoverEnabled) r.recover {
                           case NonFatal(e) =>
-                            log.warning("Task [{}] failed in phase [{}]: {}", taskName, phase, e.getMessage)
+                            log.warning("Task [{}] failed in phase [{}]: {}", names.mkString(", "), phase, e.getMessage)
```

Comment:
The primary purpose of the `taskName` was this logging, to be able to see exactly which task that failed (given that the user provided some meaningful names). Looks like that is lost now by listing all names in the phase.

Reply:
It's not all names in the phase, it's all the names under which this particular thunk was registered.